src/dcp/dcp.c

```diff
@@ -570,6 +585,39 @@ int main(int argc, char** argv)
     daos_cleanup(daos_args, mfu_src_file, mfu_dst_file);
 #endif
 
+    /* restore uid */
+    if (euid > 0) {
+        if (seteuid(uid) < 0) {
+            MFU_LOG(MFU_LOG_ERR, "Could not restore original User ID: %s", strerror(errno));
+            mfu_finalize();
+            MPI_Finalize();
+            return 1;
+        }
+        MFU_LOG(MFU_LOG_DBG, "Restored User ID back to %u", uid);
+    }
+
+    /* restore gid */
+    if (egid > 0) {
+        if (setegid(gid) < 0) {
+            MFU_LOG(MFU_LOG_ERR, "Could not restore original Group ID: %s", strerror(errno));
+            mfu_finalize();
+            MPI_Finalize();
+            return 1;
+        }
+        MFU_LOG(MFU_LOG_DBG, "Restored Group ID back to %u", gid);
+    }
+
+    /* restore groups*/
+    if (egid > 0 || euid > 0) {
+        if (setgroups(gids_count, gids) < 0) {
```

Comment:
Is gids_count ever set to something other than 0?  I didn't see that it was.

Reply:
Good catch. Thanks.